[Correction] Only margin can be changed in @page

sunkanmii opened the pull request:
Although the current description says;
_"You can only change the margins, orphans, widows, and page breaks of the document"_, but after testing, I found out that only the **margin** can be changed in the @page at-rule. 
Maybe, all the features in the official spec have not been implemented yet? But, this is a small change I think would be useful to all devs that come across this and try it out for the first time.

<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
<!-- ✍️ In a sentence or two, describe your changes -->
It changes the description that says @page can only change the margins, orphans, widows, and page breaks of the document to only the margin can be changed.

#### Motivation
<!-- ❓ Why are you making this change? Help us understand how your changes help readers. -->
I came across this at-rule because I was preparing an article for a blog, but after testing I found out that only the margin can be changed. I think other devs that come across this won't waste their time wondering why the other properties were not showing any visible effect

#### Supporting details
<!-- 🔗 Link to supporting information, such as bug trackers, source control, release notes, and vendor docs. -->
The official documentation says a lot more can be changed, but, those changes also don't have any visible effect after testing. I think implementation for those properties is yet to be done in all UAs.
This is the official doc - [https://drafts.csswg.org/css-page/#at-page-rule](https://drafts.csswg.org/css-page/#at-page-rule)

#### Related issues
<!-- 🔨 If applicable, use "Fixes #XYZ" -->
Not applicable

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

sunkanmii committed:
[Correction] Only margin can be changed in @page

Although the current description says;
_"You can only change the margins, orphans, widows, and page breaks of the document"_, but after testing, I found out that only the **margin** can be changed in the @page at-rule. 
Maybe, all the features in the official spec have not been implemented yet? But, this is a small change I think would be useful to all devs that come across this and try it out for the first time.

teoli2003 commented:
Which browsers did you test this on?

sunkanmii commented:
Chrome, Firefox, Edge, and Opera.
They are all outputting the same result.

wbamberg commented:
For my education as much as anything, can you find a reference in the spec to back up the statement in the page that:

> You can only change the margins, orphans, widows, and page breaks of the document

? I couldn't. We should document the spec, primarily, and then note ways in which browsers differ from it, usually but not always in the browser compatibility table.

teoli2003 commented:
The main spec is https://drafts.csswg.org/css-page/#at-page-rule (it was missing from mdn/browser-compat-data, but I added it. It is in the latest bcd package that will be used by Yari later today).

The spec says:
![Capture d’écran 2022-08-31 à 07 25 37](https://user-images.githubusercontent.com/1466293/187599097-c41ae297-7e5e-4dd4-80b9-7a302556b468.png)

So only margin at-rules can be added, and this PR is valid.

I guess the author has been fooled by the prose before, which shows the spec editors intend to allow more content inside `@page` in the future.

teoli2003 commented on files/en-us/web/css/@page/index.md:
```diff
@@ -36,7 +36,7 @@ The **`@page`** CSS at-rule is used to modify some CSS properties when printing
 ## Description
 
-You can't change all CSS properties with `@page`. You can only change the margins, orphans, widows, and page breaks of the document. Attempts to change any other CSS properties will be ignored.
+You can't change all CSS properties with `@page`. You can only change the margin of the document. Attempts to change any other CSS properties will be ignored.
```
```suggestion
You can't change all CSS properties with `@page`. You can only change the margins of the document. Attempts to change any other CSS properties will be ignored.
```

wbamberg commented:
> The main spec is https://drafts.csswg.org/css-page/#at-page-rule (it was missing from mdn/browser-compat-data, but I added it. It is in the latest bcd package that will be used by Yari later today).
> 
> The spec says: [Sorry, it is one of those days where the CSSWG site is broken] ![Capture d’écran 2022-08-31 à 07 25 37](https://user-images.githubusercontent.com/1466293/187599097-c41ae297-7e5e-4dd4-80b9-7a302556b468.png)
> 
> So only margin at-rules can be added, and this PR is valid if I have understood this correctly.
> 
> I guess the author has been fooled by the prose before, which shows the spec editors intend to allow more content inside `@page` in the future.

Are you sure? My read of this is that "margin at-rule" is a different thing to "margin property":

<img width="870" alt="Screen Shot 2022-08-30 at 10 55 07 PM" src="https://user-images.githubusercontent.com/432915/187604312-ff5db1cf-c7c1-46fc-9b10-9e1000fbbaea.png">

(i.e. it is a thing that describes a section of the page margin.)

teoli2003 commented:
Yes, you may be right.
@sunkanmii: what have you tested with, the properties or the at-rules?

sunkanmii commented:
I had taken my time to test the **margin at-rules** and those didn't give any visible effect either.
Only the **margin** property can be changed in the @page after testing with the browsers I listed earlier.

sunkanmii commented:
The spec was supposed to allow a wide range of features, yes, but the only property I've seen that has a visible effect amongst the listed ones on both MDN and the spec is the **margin** property.

wbamberg commented:
It looks to me as if the list is here: https://drafts.csswg.org/css-page/#page-property-list (linked from https://drafts.csswg.org/css-page/#page-properties), does that look right? But I also tried out other properties and could only get `margin` to work.

sunkanmii commented:
Yes. That looks exactly right.

sunkanmii committed:
Merge branch 'main' into patch-1

wbamberg commented:
So I suppose we should update this PR with the correct list of properties, per spec, and then also note that currently browsers only support `margin`?

There's lots of other work that should be done on this page (incomplete list of descriptors, nothing on margin boxes, no examples...), but I don't think we should block this PR on it all.

teoli2003 commented:
We probably want to update mdn/browser-compat-data  too. @queengooborg Do you know if there is a way to test these things easily?

But I agree this shouldn't block this PR.

sunkanmii commented:
If that's the case, let me also point out that the pseudo-classes (`:first`, `:left`, etc) under the `@page` at-rule have something similar going on with them.

Only the `margin` property works with them as well (I tested it, but you can also test it on the browsers I listed earlier to confirm), so if you're going to rewrite the whole at-rule again, I think removing the other properties stated in those pseudo-classes be removed, and leaving only the `margin` would remove the stress of unnecessary testing.

Also, thank you for not closing my PR! :')

sunkanmii commented:
Is it okay if I submit other pull requests specifically for `:first`, `:left`, and `:right` for the work process to have a bit of structure, or should we just leave everything **here** if we're going ahead to update the pseudo-classes mentioned?

wbamberg commented:
> Is it okay if I submit other pull requests specifically for `:first`, `:left`, and `:right` for the work process to have a bit of structure, or should we just leave everything **here** if we're going ahead to update the pseudo-classes mentioned?

Smaller, simpler PRs are more likely to get reviewed quickly but in this case I'd probably prefer it to be one PR. But honestly, it's pretty much up to you :).

sunkanmii commented:
I think it's alright if we leave it here in one PR then. So it's reviewed faster.

sunkanmii committed:
Merge branch 'main' into patch-1

estelle commented:
> Authors can specify various aspects of a page box, such as its dimensions, orientation, and margins, within an @page rule. [@page](https://drafts.csswg.org/css-page/#at-ruledef-page) rules are allowed wherever [rule-sets](https://www.w3.org/TR/CSS2/syndata.html#rule-sets) are allowed. An [@page](https://drafts.csswg.org/css-page/#at-ruledef-page) rule consists of the keyword [@page](https://drafts.csswg.org/css-page/#at-ruledef-page), an optional comma-separated list of [page selectors](https://drafts.csswg.org/css-page/#page-selector) and a block of declarations (said to be in the page context). An [@page](https://drafts.csswg.org/css-page/#at-ruledef-page) rule can also contain other at-rules, interleaved between declarations. **The current level of this specification only allows [margin at-rules](https://drafts.csswg.org/css-page/#margin-at-rule) inside [@page](https://drafts.csswg.org/css-page/#at-ruledef-page).**

Last sentence is my hilite.

The original mdn content makes it sound like widow and orphans are properties that might have been configurable, but they are page break properties, so are places that can be impacted by margin (and, in the future, other CSS properties), but are not a property you would alter in @page now or in the future.

I think the whole page needs to be worked on.

An example like the following would be helpful:

```css 
@page :first {
  margin-top: 4in;
}
```

the page has two descriptors (orientation and size), but no example of how either is used. 

To test to see if other properties work, how about trying:

```css
@page {
  @top-right {
    content: "Content works";
  }
}
```

not sure why the spec is pointing to https://drafts.csswg.org/css-logical/#page/. It should point to https://drafts.csswg.org/css-page/#at-page-rule

wbamberg commented:
> > Authors can specify various aspects of a page box, such as its dimensions, orientation, and margins, within an @page rule. [@page](https://drafts.csswg.org/css-page/#at-ruledef-page) rules are allowed wherever [rule-sets](https://www.w3.org/TR/CSS2/syndata.html#rule-sets) are allowed. An [@page](https://drafts.csswg.org/css-page/#at-ruledef-page) rule consists of the keyword [@page](https://drafts.csswg.org/css-page/#at-ruledef-page), an optional comma-separated list of [page selectors](https://drafts.csswg.org/css-page/#page-selector) and a block of declarations (said to be in the page context). An [@page](https://drafts.csswg.org/css-page/#at-ruledef-page) rule can also contain other at-rules, interleaved between declarations. **The current level of this specification only allows [margin at-rules](https://drafts.csswg.org/css-page/#margin-at-rule) inside [@page](https://drafts.csswg.org/css-page/#at-ruledef-page).**
> 
> Last sentence is my hilite.
> 

As I said in https://github.com/mdn/content/pull/20012#issuecomment-1232495558, I don't think that "margin at-rules" are the same thing as "margin properties", and I thought this issue is about margin properties.

(I agree that this page needs lots more work.)

sunkanmii commented:
> > Authors can specify various aspects of a page box, such as its dimensions, orientation, and margins, within an @page rule. [@page](https://drafts.csswg.org/css-page/#at-ruledef-page) rules are allowed wherever [rule-sets](https://www.w3.org/TR/CSS2/syndata.html#rule-sets) are allowed. An [@page](https://drafts.csswg.org/css-page/#at-ruledef-page) rule consists of the keyword [@page](https://drafts.csswg.org/css-page/#at-ruledef-page), an optional comma-separated list of [page selectors](https://drafts.csswg.org/css-page/#page-selector) and a block of declarations (said to be in the page context). An [@page](https://drafts.csswg.org/css-page/#at-ruledef-page) rule can also contain other at-rules, interleaved between declarations. **The current level of this specification only allows [margin at-rules](https://drafts.csswg.org/css-page/#margin-at-rule) inside [@page](https://drafts.csswg.org/css-page/#at-ruledef-page).**
> 
> Last sentence is my hilite.
> 
> The original mdn content makes it sound like widow and orphans are properties that might have been configurable, but they are page break properties, so are places that can be impacted by margin (and, in the future, other CSS properties), but are not a property you would alter in @page now or in the future.
> 
> I think the whole page needs to be worked on.
> 
> An example like the following would be helpful:
> 
> ```css
> @page :first {
>   margin-top: 4in;
> }
> ```
> 
> the page has two descriptors (orientation and size), but no example of how either is used.
> 
> To test to see if other properties work, how about trying:
> 
> ```css
> @page {
>   @top-right {
>     content: "Content works";
>   }
> }
> ```
> 
> not sure why the spec is pointing to https://drafts.csswg.org/css-logical/#page/. It should point to https://drafts.csswg.org/css-page/#at-page-rule

I and @wbamberg previously mentioned that we individually tested both the `margin at-rules` (```@top-right{}, @top-left{}, etc```), with all the CSS properties in the [spec](https://drafts.csswg.org/css-page/#page-property-list) and we found out that none of the `margin at-rules` worked and only the `margin` CSS property had a visible effect.

I think @wbamberg 's [comment](https://github.com/mdn/content/pull/20012#issuecomment-1233245433) would work best on the updates to be made on @page, to avoid confusion later on with other developers that come across the `@page at-rule`.

sunkanmii committed:
Merge branch 'main' into patch-1

estelle committed:
Merge branch 'main' into patch-1

wbamberg commented:
@sunkanmii , thanks for coming back to this PR!

If you would like a review, please update the PR so we can see the actual changes proposed.

In addition it's a good idea to explain what intended changes are. For example, I was expecting something like the changes listed in https://github.com/mdn/content/pull/20012#issuecomment-1233245433:

> the correct list of properties, per spec, and then also note that currently browsers only support margin

From a look at your dropbox it seems like there are lots more changes! If you want to add extra changes in this PR, then that might be OK, but please explain specifically what they are.

sunkanmii commented:
@wbamberg Ah. sorry about that!

The changes I intend to make from the dropbox link are:
* The at-rule definition, I thought it was a bit confusing as to "some CSS properties" from the definition over here
> The @page CSS at-rule is used to modify some CSS properties when printing a document.

and I thought it might be better to give one based on the official w3.org doc

* Also changed the **Syntax** based on [estelle's comment](https://github.com/mdn/content/pull/20012#issuecomment-1238449538) using the `:left` and `:right` pseudo-classes instead of just `:first`, the **Descriptor** to match what's currently supported, and note that only the `margin` property is currently the only CSS property supported.

* And I basically imported and explained the **Formal Syntax**, what properties are intended to work, and I closed it with a live example

I may have just misunderstood the last part of your comment in [#20012](https://github.com/mdn/content/pull/20012#issuecomment-1233245433:)
> There's lots of other work that should be done on this page (incomplete list of descriptors, nothing on margin boxes, no examples...), but I don't think we should **block this PR on it all**.

I thought it was a typo and you meant not blocking my PR because of the work to be done. My bad.

wbamberg commented:
> 
> > There's lots of other work that should be done on this page (incomplete list of descriptors, nothing on margin boxes, no examples...), but I don't think we should **block this PR on it all**.
> 
> I thought it was a typo and you meant not blocking my PR because of the work to be done. My bad.

Sorry to be unclear, @sunkanmii . What I meant by "I don't think we should block this PR on it all" is: we should keep this PR focused on the original issue and not add lots of extra stuff to it.

For BCD: that's maintained in a separate repo, https://github.com/mdn/browser-compat-data. Specifically you would be editing the JSON file at https://github.com/mdn/browser-compat-data/blob/main/css/at-rules/page.json. MDN pages read in these JSON files and render them in a table.

sunkanmii committed:
Merge branch 'main' into patch-1

bsmth committed:
Update files/en-us/web/css/@page/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

bsmth committed:
Merge branch 'main' into patch-1

sunkanmii committed:
Updated @page to match current documentation

Here's a summary of the changes I made:
* The at-rule definition. I thought it was a bit confusing as to "some CSS properties" from the definition over here
> The @page CSS at-rule is used to modify some CSS properties when printing a document.

and I thought it might be better to give one based on the official w3.org doc

* Also changed the Syntax based on estelle's comment using the `:left` and `:right` pseudo-classes instead of just `:first`, the **Descriptor** to match what's currently supported, and note that only the margin property is presently the only CSS property supported.

* Removed properties and pseudo-classes that had no links to them or were not accurate with what was in the documentation.

* And I imported and explained the Formal Syntax, and what properties are intended to work, and I closed it with a live example

estelle committed:
Merge branch 'main' into patch-1

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+
+### page-properties
+
+| bidi properties       | direction             |
+| --------------------- | --------------------- |
```
```suggestion
| Feature     | CSS properties             |
| --------------------- | --------------------- |
| bidi properties       | direction             |
```

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+|                       | border                |
+| counter properties    | counter-reset         |
+|                       | counter-increment     |
+| color                 |                       |
```
```suggestion
| color                 |     color                  |
```

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+| height properties     | height                |
+|                       | min-height            |
+|                       | max-height            |
+| line-height           |                       |
```
```suggestion
| line height           |          line-height             |
```

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+|                       | padding-bottom        |
+|                       | padding-left          |
+|                       | padding               |
+| quotes                |                       |
```
```suggestion
| quotes                |           quotes            |
```

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+|                       | text-transform        |
+|                       | white-space           |
+|                       | word-spacing          |
+| visibility            |                       |
```
```suggestion
| visibility            |       visibility                |
```

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+    });
+```
+Print out the page below to see the output of this example:
+%[https://codepen.io/sunkanmii-the-styleful/pen/dyezNry]
````
Codepen uses command+p to open a new pen, so thought using https://codepen.io/sunkanmii-the-styleful/full/dyezNry would work, but that prints all of codepen, not what you're looking for. 

We should remove the codepen link and either do a live-example, which I think will work, or do a full web page (not codepen) and add an interactive-example page.

sunkanmii replied:
Removed the codepen link and added a live-example.

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
 
 ```
-@page <page-selector-list> {
+@page <pseudo-selector> {
````
none of the examples use the size descriptor. Weould be good if we showed an example of that.

sunkanmii replied:
The examples at the start actually do not, would change that.
I'll add the size descriptor to two of the examples above.

sunkanmii replied:
Added the size to the first example

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+- {{Cssxref(":right")}}
+
+
+> Note: The **`@page`** at-rule can only contain **page-properties** and **margin at-rules**, and the **margin at-rules** can only contain **page-margin properties**
```
```suggestion
> **Note:** The **`@page`** at-rule can only contain **page-properties** and **margin at-rules**, and the **margin at-rules** can only contain **page-margin properties**
```

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+      <button>Print Webpage</button>
+    </body>
+```
+Relevant CSS code:
+```
+    @page {
+      size: landscape;
+      margin: 20%;
+    }
+```
+
+JS:
+```
+    let button = document.querySelector("button")
````
```suggestion

Relevant CSS code:
```css
    @page {
      size: landscape;
      margin: 20%;
    }
```

JavaScript:

```js
    let button = document.querySelector("button")
```

estelle replied:
```suggestion
    let button = document.querySelector("button");
```

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+    </body>
+```
+Relevant CSS code:
+```
````
```suggestion

```css
```

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+    }
+```
+
+JS:
+```
````
```suggestion
JavaScript

```js
```

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+```
+#### page-margin properties
+
+| bidi properties       | direction             |
````
The same edits for the previous table apply to this one as well

sunkanmii replied:
Done.

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,12 +51,282 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+```
+    @page {
+      size: landscape;
+      margin: 20%;
````
Here you say relevant CSS, but in the codepen you include   
```
section {
page-break-after: always;
  break-after: page;
  }
``` 
which is causing 5 page breaks, but doesn't appear here so will confuse the reader.

sunkanmii replied:
That's true. Would add that.

sunkanmii replied:
So because we needed a live embed, I had to add the whole CSS code.

sunkanmii replied:
It also contains the `section` CSS code that was missing.

sunkanmii committed:
applied bold text to second "Note:"

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
added table titles to page-properties

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
added color to Feature on page-properties

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
added line-height to Features on page-properties

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
added quotes to Features on page-properties

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
added correct markup for CSS code

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
added visibility to Features on page-properties

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
added correct markup for JavaScript code

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
semicolon added to end of JavaScript query code

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
Added changes mentioned by @estelle

* Added size descriptor to the first example
* Added table title to page-margin properties section table
* Added live sample embed for the last example

sunkanmii commented:
@estelle Thanks a lot for editing and pointing out the things I missed.
I've applied the changes you mentioned.

Awaiting your feedback!

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -14,32 +14,37 @@ browser-compat: css.at-rules.page
 ## Description
-
-You can't change all CSS properties with `@page`. You can only change the margins, orphans, widows, and page breaks of the document. Attempts to change any other CSS properties will be ignored.
+> Note: This documentation lists out features yet to be implemented across UAs, but they are stated in the official documentation. At the time of this writing, the only CSS property shown to have visible on both the at-rule and its pseudo-classes is the `margin` property.
```
```suggestion
> **Note:** This documentation lists out features yet to be implemented across UAs, but they are stated in the official documentation. At the time of this writing, the only CSS property shown to have visible on both the at-rule and its pseudo-classes is the `margin` property.
```
The asterisks are needed for this to appear as a note. Othewise it appears as a pullquote.

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,12 +53,314 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+|                       | border                |
+| counter properties    | counter-reset         |
+|                       | counter-increment     |
+| content               |                       |
+| color                 |                       |
```
```suggestion
| content               |         content              |
| color                 |        color               |
```

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,12 +53,314 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+| height properties     | height                |
+|                       | min-height            |
+|                       | max-height            |
+| line-height           |                       |
```
```suggestion
| line height           |        line-height               |
```

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,12 +53,314 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+|                       | padding-bottom        |
+|                       | padding-left          |
+|                       | padding               |
+| quotes                |                       |
+| text properties       | letter-spacing        |
+|                       | text-align            |
+|                       | text-decoration       |
+|                       | text-indent           |
+|                       | text-transform        |
+|                       | white-space           |
+|                       | word-spacing          |
+| vertical-align        |                       |
+| visibility            |                       |
+| width properties      | width                 |
+|                       | min-width             |
+|                       | max-width             |
+| z-index               |                       |
+
```
```suggestion
| quotes                |        quotes               |
| text properties       | letter-spacing        |
|                       | text-align            |
|                       | text-decoration       |
|                       | text-indent           |
|                       | text-transform        |
|                       | white-space           |
|                       | word-spacing          |
| vertical alignment        |          vertical-align             |
| visibility            |        visibility               |
| width properties      | width                 |
|                       | min-width             |
|                       | max-width             |
| z-index               |         z-index              |

```

sunkanmii committed:
asterisks added to first note under Description

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
added `color` and `content` to page-margin properties content

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
`line-height` added to page-margin properties content table

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
quotes, vertical-align, visibility, and z-index corrected on page-margin properties table

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

estelle committed:
Merge branch 'main' into patch-1

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -14,32 +14,37 @@ browser-compat: css.at-rules.page
 {{CSSRef}}
 
-The **`@page`** CSS at-rule is used to modify some CSS properties when printing a document.
+The **`@page`** at-rule is a CSS at-rule used to modify different aspects of a printed page property. It targets and modifies the page’s dimensions, page orientation, and margins. The **`@page`** at-rule can be used to target all pages in a print-out, or even specific ones using its various pseudo-classes.
```
```suggestion
The **`@page`** at-rule is a CSS at-rule used to modify different aspects of a printed page property. It targets and modifies the page's dimensions, page orientation, and margins. The **`@page`** at-rule can be used to target all pages in a print-out, or even specific ones using its various pseudo-classes.
```

estelle committed:
curly quote removal

estelle committed:
trying to get example to work

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,12 +53,313 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+|                       | min-width             |
+|                       | max-width             |
+
+### margin-at rules
+The margin-at rules are used inside of the `@page` at-rule and they target specific sections of a page document and make changes based on the properties or content set in the block of code i.e 
+```
+    @page {
````
```suggestion

### margin-at rules

The margin-at rules are used inside of the `@page` at-rule and they target specific sections of a page document and make changes based on the properties or content set in the block of code:

```
    @page {
```

estelle committed:
line spacing

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,13 +53,316 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+      }
+    }
+```
+**`@top-left`** targets the top-left of the document and applies the changes based on the page-margin-properties set.
+
+Other margin-at rules include:
````
```suggestion

**`@top-left`** targets the top-left of the document and applies the changes based on the page-margin-properties set.

Other margin-at rules include:

```

estelle committed:
line spacing

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,13 +53,318 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
 
+The example below prints out 5 sections different sections in `landscape` size, and adds a 20% margin to each of them in the print-out.
+
+HTML:
```
```suggestion
HTML:

```

estelle replied:
spacing

estelle committed:
Update files/en-us/web/css/@page/index.md

estelle committed:
linting issues

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -14,32 +14,37 @@ browser-compat: css.at-rules.page
 ## Description
-
-You can't change all CSS properties with `@page`. You can only change the margins, orphans, widows, and page breaks of the document. Attempts to change any other CSS properties will be ignored.
+> **Note:** This documentation lists out features yet to be implemented across UAs, but they are stated in the official documentation. At the time of this writing, the only CSS property shown to have visible on both the at-rule and its pseudo-classes is the `margin` property.
```
```suggestion

> **Note:** This documentation lists out features yet to be implemented across UAs, but they are stated in the official documentation. At the time of this writing, the only CSS property shown to have visible on both the at-rule and its pseudo-classes is the `margin` property.
```

estelle committed:
line space

estelle commented on files/en-us/web/css/@page/index.md:
````diff
@@ -48,13 +55,320 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+```
+
+#### page-margin properties
+
````
Add a sentence here explaining what these are. Maybe something like "When printing, the set of properties that can be set on any individual margin at-rule include:" or something similar. It's not clear why this second list of properties is here.

sunkanmii replied:
That makes sense. Would do that.

sunkanmii replied:
Done.

estelle commented on files/en-us/web/css/@page/index.md:
```diff
@@ -48,13 +55,320 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
+  size: landscape;
+  margin: 20%;
+}
+
```
```suggestion
@page :left {
   @top-right {
       content: "⎗";
    }
}
```

estelle replied:
Thinking it would be cool to add a second example. Not sure if this example is the best, but something would be cool,

estelle replied:
Ah, this doesn't work yet. Let me ask others if they think we should add a non-working example with a comment, or omit it. I think it would be nice to keep it with a comment, but not sure that is how we do it at MDN.

sunkanmii replied:
Hmmm. I've been reading your docs for a while now and I have seen some non-working examples like the one in :local-link.
It's just that we would need to show what the result would look like, and the documentation itself doesn't state how it would look like so I didn't even bother.

sunkanmii replied:
I have experimented with it countless of times, and when I noticed it didn't work, I didn't bother writing an example.

estelle reviewed (commented):
Three more comments, and i think we may be good to go.

sunkanmii committed:
modified margin at-rules definition

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sunkanmii committed:
Added description for page-margin properties

sunkanmii commented:
Hey, @estelle 
Any updates so far?

sunkanmii commented:
Thanks again for all your help!
Awaiting your feedback.

sunkanmii committed:
Merge branch 'main' into patch-1

estelle reviewed (approved):
Yay! Thank you for writing this.

estelle committed:
Merge branch 'main' into patch-1

sunkanmii commented:
Yay! Thank you for merging my PR. :)
